File: src/main/java/com/dh/ondot/member/application/command/OnboardingCommand.java
package com.dh.ondot.member.application.command;

import com.dh.ondot.member.api.request.OnboardingRequest;

public record OnboardingCommand(
        Integer preparationTime,
        String alarmMode,
        Boolean isSnoozeEnabled,
        Integer snoozeInterval,
        Integer snoozeCount,
        String soundCategory,
        String ringTone,
        Double volume
) {
    public static OnboardingCommand from(OnboardingRequest request) {
        return new OnboardingCommand(
                request.preparationTime(),
                request.alarmMode(),
                request.isSnoozeEnabled(),
                request.snoozeInterval(),
                request.snoozeCount(),
                request.soundCategory(),
                request.ringTone(),
                request.volume()
        );
    }
}

File: src/main/kotlin/com/dh/ondot/member/application/command/OnboardingCommand.kt
package com.dh.ondot.member.application.command

data class OnboardingCommand(
    val preparationTime: Int,
    val alarmMode: String,
    val isSnoozeEnabled: Boolean,
    val snoozeInterval: Int,
    val snoozeCount: Int,
    val soundCategory: String,
    val ringTone: String,
    val volume: Double,
)
